### ISSUE TITLE
MACD Result having confusing property names

### ISSUE THREAD

**Issue by crixx** [bennycode/trading-signals#165]
Not sure, but I think `diff` and `macd` are switched in the MACD Result: 
````js
    /**
     * The MACD-Histogram represents the difference between MACD and its 9-day EMA, the signal line.
     */
    this.result = {
      diff: diff,
      macd: diff.sub(this.signal.getResult()),
      signal: this.signal.getResult(),
    };
````

From my understanding, the `diff` should actually be the difference between the `macd` and the `signal`, where the `macd` is the differnce between the two EMAs... What do you think @bennycode ?

**Comment by bennycode** [comment]
Hi @crixx and thank you for your report! Do you have a reference from another analysis library (e.x. pandas for Python)?

**Comment by crixx** [comment]
Hi @bennycode, actually not no. But just from the jsdoc:
````
     /**
     * The MACD-Histogram represents the difference between MACD and its 9-day EMA, the signal line.
     */
````
I'd say `diff = macd - signal`, and therefore `diff = (shortEMA-longEma)-signal`

The jsdoc some lines above sounds pretty the same:
````
    /**
     * Standard MACD is the short (usually 12 periods) EMA less the long (usually 26 periods) EMA. Closing prices are
     * used to form the moving averages.
     */
    const diff = shortEMA.sub(longEMA);
````
There I'd expect to find `const macd = shortEMA.sub(longEMA);`


For me it's fine as it is, just wanted to let you know this was something I lost some hours debugging...

**Comment by bennycode** [comment]
Hey @crixx, thanks again for getting back to me!

Today, I had the time to verify the MACD results with other libraries and you have been totally right, `diff` and `macd` have been switched. 😲

From what I understood, it must be as follows:

- **MACD:** Short EMA - Long EMA
- **Histogram:** MACD - Signal

To avoid confusion, I removed `diff` and added `histogram`. I also made sure that `macd` reports now the values as you suggested. I matched my values with the [Example Calculation](https://tulipindicators.org/macd) from Tulip Indicators.

I also changed the API contract. You have to supply now a constructor in your MACD config, that will be used to create the signal indicator. 

**Example:**

```ts
import {EMA, MACD} from 'trading-signals';

const macd = new MACD({longInterval: 26, shortInterval: 12, signalInterval: 9, indicator: EMA});
```

All of the above is available in **trading-signals v1.3.0**.

Thank you very much for your support!
